manifests/appdefaults.pp

```diff
@@ -12,26 +12,35 @@
 # Copyright 2013 Patrick Mooney.
 # Copyright (c) IN2P3 Computing Centre, IN2P3, CNRS
 #
-
-define mit_krb5::appdefaults(
-  $debug                 = '',
-  $ticket_lifetime       = '',
-  $renew_lifetime        = '',
-  $forwardable           = '',
-  $krb4_convert          = '',
-  $ignore_afs            = '',
+# @param debug
+#
+# @param ticket_lifetime
+#
+# @param renew_lifetime
+#
+# @param forwardable
+#
+# @param krb4_convert
+#
+# @param ignore_afs
+#
+define mit_krb5::appdefaults (
+  Optional[String] $debug           = undef,
+  Optional[String] $ticket_lifetime = undef,
+  Optional[String] $renew_lifetime  = undef,
+  Optional[String] $forwardable     = undef,
```

Comment:
To ensure backward compatibility, I think that we should introduce a new _data type_ `Mit_krb5::Bool_or_str` such as:

```puppet
type Mit_krb5::Bool_or_str = Variant[Boolean, Enum['true', 'false']]
```

Once introduced, `$forwadable` should be of type `Optional[Mit_krb5::Bool_or_str]`